builder/actions/setup_ci_from_files.py

```diff
@@ -0,0 +1,301 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+from builder.core.action import Action
+import os
+import pathlib
+import sys
+import json
+import tempfile
+import xml.etree.cElementTree as xml
+
+
+class SetupCIFromFiles(Action):
+    current_folder = None
+    env_instance = None
+    tmp_file_storage = []  # NOTE: This is needed to keep the tmp files alive
+
+    def _is_same_os_as_string(self, os_string):
```

Comment:
most of the logic here seems to be the same as current_os helper we already have in host.py